lib/bundler/metrics.rb

```diff
@@ -0,0 +1,237 @@
+      @system_metrics["gem_source_count"] = Bundler.definition.sources.rubygems_sources.count
+      require "digest"
+      # hashes each gem source domain (ex. https://rubygems.org/)
+      @system_metrics["gem_sources"] = Bundler.definition.sources.rubygems_sources.map(&:to_s).map {|source| Digest::MD5.hexdigest(source[source.index(/http/)..source.rindex("/")]) if source.match(/http/) }
+    end
+
+    def self.send_metrics
+      begin
+        uri = URI "https://rubygems.org/api/metrics"
+        # use persistent connection to so we can use a single connection to rubygems
+        require_relative "vendored_persistent"
+        http = Bundler::Persistent::Net::HTTP::Persistent.new
+        post = Net::HTTP::Post.new(uri)
+        # YAML format is required to properly access the keys in the backend API, as we cant "require json" in bundler
+        post["Content-Type"] = "application/vnd.yaml"
+        require "yaml"
+        post.body = read_from_file.to_yaml
+        http.request(uri, post)
+      rescue Bundler::Persistent::Net::HTTP::Persistent::Error, Errno::ENETUNREACH, Errno::ENOENT
+        "Connection failed"
+      rescue OpenSSL::SSL::SSLError
+        "Certification has probably expired"
+      end
+      # The file is emptied after sending metrics
+      # File::TRUNC is preferable since File.truncate doesn't work for all systems
+      begin
+        open(@path, File::TRUNC) if File.exist?(@path)
+      rescue Errno::EINVAL # CI's fail here otherwise
+      end
+    end
+
+    def self.git_info
+      begin
+        origin = `git remote get-url origin`
+        origin = `git config --get remote.origin.url` if origin.empty? # for older git versions
+        require "digest"
+        # hash the origin repository to calculate the number of unique bundler users
+        @system_metrics["origin"] = Digest::MD5.hexdigest(origin.chomp) unless origin.empty?
+      rescue Errno::ENOENT
+      end
+      begin
+        git_ver = `git --version`
+        @system_metrics["git_version"] = git_ver[git_ver.index(/\d/)..git_ver.rindex(/\d/)].chomp unless git_ver.empty?
+      rescue Errno::ENOENT
+      end
+    end
+
+    def self.ruby_env_managers
+      # TODO: FIX THIS ONCE RVM --VERSION STOPS THROWING ERRORS
+      # this is commented out because RVM may throw the following, when running any RVM command:
+
+      # Warning! PATH is not properly set up, /workspace/.rvm/bin is not at first place.
+      #    Usually this is caused by shell initialization files. Search for PATH=... entries.
+      #    You can also re-add RVM to your profile by running: rvm get stable --auto-dotfiles
+      #    To fix it temporarily in this shell session run: rvm use .rvm
+      #    To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.
+
+      # this results in test failures in tests which don't expect errors.
+      # the command DOES run successfully!
+      # begin
+      #   rvm_ver = `rvm --version`
```

Comment:
The issue lied here.
After setting up RVM on a gitpod.io system, running any RVM command ran the command successfully, and also echoed:
``` 
Warning! PATH is not properly set up, /workspace/.rvm/bin is not at first place.
      Usually this is caused by shell initialization files. Search for PATH=... entries.
      You can also re-add RVM to your profile by running: rvm get stable --auto-dotfiles
      To fix it temporarily in this shell session run: rvm use .rvm
      To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.
```
This is what caused the tests to fail.
The way I see this, we either:
1. Dont collect user RVM version, or:
2. We change all the failing tests to accept this warning.